imap_processing/spice/pointing_frame_handler.py

```diff
@@ -0,0 +1,242 @@
+        Array of times between et_start and et_end.
+
+    Returns
+    -------
+    rotation_matrix : np.ndarray
+        Rotation matrix.
+    """
+    # Averaged quaternions.
+    q_avg = average_quaternions(et_times)
+
+    # Converts the averaged quaternion (q_avg) into a rotation matrix
+    # and get inertial z axis.
+    # https://spiceypy.readthedocs.io/en/main/documentation.html#spiceypy.spiceypy.q2m
+    z_avg = spice.q2m(list(q_avg))[:, 2]
+    # y_avg is perpendicular to both z_avg and the standard Z-axis.
+    y_avg = np.cross(z_avg, [0, 0, 1])
+    # x_avg is perpendicular to y_avg and z_avg.
+    x_avg = np.cross(y_avg, z_avg)
+
+    # Construct the rotation matrix from x_avg, y_avg, z_avg
+    rotation_matrix = np.asarray([x_avg, y_avg, z_avg])
+
+    return rotation_matrix
+
+
+def create_pointing_frame() -> Path:
+    """
+    Create the pointing frame.
+
+    Returns
+    -------
+    path_to_pointing_frame : Path
+        Path to dps frame.
+    """
+    # TODO: this part will change with ensure_spice decorator.
+    # Mount path to EFS.
+    mount_path = Path(os.getenv("EFS_MOUNT_PATH", ""))
+
+    # TODO: this part will change with ensure_spice decorator.
+    kernels = [str(file) for file in mount_path.iterdir()]
+    ck_kernel = [str(file) for file in mount_path.iterdir() if file.suffix == ".bc"]
+
+    # Furnish the kernels.
+    with spice.KernelPool(kernels):
+        # Get timerange for the pointing frame kernel.
+        et_start, et_end, et_times = get_et_times(str(ck_kernel[0]))
+        # Create a rotation matrix
+        rotation_matrix = create_rotation_matrix(et_times)
+
+        # Convert the rotation matrix to a quaternion.
+        # https://spiceypy.readthedocs.io/en/main/documentation.html#spiceypy.spiceypy.m2q
+        q_avg = spice.m2q(rotation_matrix)
+
+        # TODO: come up with naming convention.
+        path_to_pointing_frame = mount_path / "imap_dps.bc"
+
+        # Open a new CK file, returning the handle of the opened file.
+        # https://spiceypy.readthedocs.io/en/main/documentation.html#spiceypy.spiceypy.ckopn
+        handle = spice.ckopn(str(path_to_pointing_frame), "CK", 0)
+        # Get the SCLK ID.
+        # https://spiceypy.readthedocs.io/en/main/documentation.html#spiceypy.spiceypy.gipool
+        id_imap_sclk = spice.gipool("CK_-43000_SCLK", 0, 1)
+        # https://spiceypy.readthedocs.io/en/main/documentation.html#spiceypy.spiceypy.sce2c
+        # Convert start and end times to SCLK.
+        sclk_begtim = spice.sce2c(int(id_imap_sclk), et_start)
+        sclk_endtim = spice.sce2c(int(id_imap_sclk), et_end)
+
+        # Get the pointing frame ID.
+        # https://spiceypy.readthedocs.io/en/main/documentation.html#spiceypy.spiceypy.gipool
+        id_imap_dps = spice.gipool("FRAME_IMAP_DPS", 0, 1)
+
+        # Create the pointing frame kernel.
+        # https://spiceypy.readthedocs.io/en/main/documentation.html#spiceypy.spiceypy.ckw02
+        spice.ckw02(
```

Comment:
Cool! Nice job using the lower level function rather than the executable. I missed this on my first pass.